src/main/java/org/aksw/simba/lemming/metrics/single/edgemanipulation/EdgeModification.java

```diff
@@ -258,6 +275,16 @@ public void removeEdgeFromGraph(int edgeId, int newNodeTriangles, int newEdgeTri
         this.newNodeTriangles = newNodeTriangles;
         this.newEdgeTriangles = newEdgeTriangles;
         this.graph.removeEdge(edgeId);
+        
+        //For removing an edge from a graph, update the vertex in and out degrees.
+        IntSet verticesConnectedToRemovingEdge = graph.getGraph().getVerticesIncidentToEdge(edgeId);
+        int headId = verticesConnectedToRemovingEdge.size() > 1 ? verticesConnectedToRemovingEdge.toIntArray()[1] 
+        		: verticesConnectedToRemovingEdge.toIntArray()[0];
+        int tailId = verticesConnectedToRemovingEdge.toIntArray()[0];
```

Comment:
wouldn't be something like this easier and faster?
```java
int tail = getDirectedSimpleEdgeTail(e);
int head = getDirectedSimpleEdgeHead(e);
```

Reply:
I have implemented the suggested changes.